tests/phpunit/tests/post/UpdateTermCountOnTransitionPostStatus.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+/**
+ * Tests for the _update_term_count_on_transition_post_status function.
+ *
+ * @group taxonomy
+ *
+ * @covers ::_update_term_count_on_transition_post_status
+ */
+class Tests_Taxonomy_UpdateTermCountOnTransitionPostStatus extends WP_UnitTestCase {
+
+	/**
+	 * @var int Post ID.
+	 */
+	protected $post_id;
+
+	/**
+	 * @var int Term ID.
+	 */
+	protected $term_id;
+
+	/**
+	 * @var string Post type.
+	 */
+	protected $post_type = 'post';
+
+	/**
+	 * @var string Taxonomy name.
+	 */
+	protected $taxonomy = 'category';
+
+	/**
+	 * Set up.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+
+		register_post_type( $this->post_type, array( 'public' => true ) );
+		register_taxonomy( $this->taxonomy, $this->post_type, array( 'public' => true ) );
+
+		$this->post_id = self::factory()->post->create(
+			array(
+				'post_type'   => $this->post_type,
+				'post_status' => 'publish',
+			)
+		);
+
+		$this->term_id = self::factory()->term->create(
+			array(
+				'taxonomy' => $this->taxonomy,
+				'name'     => 'Test Category',
+			)
+		);
+
+		wp_set_object_terms( $this->post_id, $this->term_id, $this->taxonomy );
+	}
```

Comment:
These can be shared fixtures so they don't need to be recreated be tween each test. Any changes within tests will be reset between test runs.

While updating, please check if you need to register the taxonomy and post type here too. I haven't done so.

```suggestion
	/**
	 * Create shared fixtures.
	 */
	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
		self::$post_id = $factory->post->create(
			array(
				'post_type'   => self::$post_type,
				'post_status' => 'publish',
			)
		);

		self::$term_id = $factory->term->create(
			array(
				'taxonomy' => self::$taxonomy,
				'name'     => 'Test Category',
			)
		);

		wp_set_object_terms( self::$post_id, self::$term_id, self::$taxonomy );
	}
```